File: src/main/java/com/kairlec/dao/MimeMapMapper.java
package com.kairlec.dao;


import com.kairlec.pojo.MimeMap;
import org.apache.ibatis.annotations.*;

import java.util.List;

@Mapper
public interface MimeMapMapper {

    Integer init();

    List<MimeMap> getAll();

    MimeMap getMimeMap(@Param("ext") String ext);

    List<MimeMap> getSection(@Param("exts") List<String> exts);

    Integer insertMimeMap(MimeMap mimeMap);

    Integer deleteMimeMap(@Param("ext") String ext);

    Integer updateMimeMap(MimeMap mimeMap);

}

File: src/main/java/com/kairlec/dao/MimeMapMapper.kt
package com.kairlec.dao


import com.kairlec.pojo.MimeMap
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface MimeMapMapper {
    fun init(): Int?
    fun getAll(): List<MimeMap>?
    fun getMimeMap(@Param("ext") ext: String): MimeMap?
    fun getSection(@Param("exts") exts: List<String>): List<MimeMap>?
    fun insertMimeMap(mimeMap: MimeMap): Int?
    fun deleteMimeMap(@Param("ext") ext: String): Int?
    fun updateMimeMap(mimeMap: MimeMap): Int?
}
